Source/wave.cpp

```diff
@@ -2,311 +2,221 @@
 
 #include "../types.h"
 
-int wave_cpp_init_value; // weak
+static float wave_cpp_init_value = INFINITY;
 
-const int wave_inf = 0x7F800000; // weak
-
-struct wave_cpp_init
-{
-	wave_cpp_init()
-	{
-		wave_cpp_init_value = wave_inf;
-	}
-} _wave_cpp_init;
-// 4802D4: using guessed type int wave_inf;
-// 6ABB34: using guessed type int wave_cpp_init_value;
-
-bool __fastcall WCloseFile(void *file)
+BOOL __fastcall WCloseFile(HANDLE file)
 {
 	return SFileCloseFile(file);
 }
 
-int __fastcall WGetFileSize(HANDLE hsFile, unsigned long *a2)
+LONG __fastcall WGetFileSize(HANDLE hsFile, DWORD *lpFileSizeHigh)
 {
-	unsigned long *v2; // edi
-	HANDLE i; // esi
-	int result; // eax
-	int a2a; // [esp+8h] [ebp-4h]
-
-	a2a = 0;
-	v2 = a2;
-	for ( i = hsFile; ; WGetFileArchive(i, &a2a, 0) )
-	{
-		result = SFileGetFileSize(i, v2);
-		if ( result )
-			break;
-	}
-	return result;
+	unsigned int retry=0;
+	LONG ret;
+
+	while ((ret = SFileGetFileSize(hsFile, lpFileSizeHigh)) == 0)
+		WGetFileArchive(hsFile, &retry, NULL);
+
+	return ret;
 }
 
-void __fastcall WGetFileArchive(HANDLE hsFile, int *a2, char *dwInitParam)
+void __fastcall WGetFileArchive(HANDLE hsFile, unsigned int *retries, const char *FileName)
 {
-	int *v3; // esi
-	HANDLE v4; // edi
-	//int v5; // eax
-	//int v6; // eax
-	HANDLE archive; // [esp+8h] [ebp-4h]
-
-	v3 = a2;
-	v4 = hsFile;
-	if ( (unsigned int)*a2 >= 5 )
-		FileErrDlg(dwInitParam);
-	if ( v4 && SFileGetFileArchive(v4, &archive) && archive != diabdat_mpq )
-	{
-		Sleep(0x14u);
-		++*v3;
-	}
-	else
+	HANDLE archive;
+
+	if (*retries >= 5)
+		FileErrDlg(FileName);
+
+	if (hsFile && SFileGetFileArchive(hsFile, &archive) && archive != diabdat_mpq)
 	{
-		//_LOBYTE(v6) = InsertCDDlg();
-		if ( !InsertCDDlg() )
-			FileErrDlg(dwInitParam);
+		Sleep(20);
+		(*retries)++;
 	}
+	else if (!InsertCDDlg())
+		FileErrDlg(FileName);
+
 }
 
-int __fastcall WOpenFile(char *dwInitParam, HANDLE *phsFile, int a3)
+BOOL __fastcall WOpenFile(const char *FileName, HANDLE *phsFile, BOOL mayNotExist)
 {
-	HANDLE *v3; // edi
-	char *i; // esi
-//	int v5; // eax
-	int a2a; // [esp+8h] [ebp-4h]
-
-	a2a = 0;
-	v3 = phsFile;
-	for ( i = dwInitParam; ; WGetFileArchive(0, &a2a, i) )
-	{
-		//_LOBYTE(v5) = SFileOpenFile(i, v3);
-		if ( SFileOpenFile(i, v3) )
-			return 1;
-		if ( a3 && SErrGetLastError() == 2 )
+	unsigned int retry = 0;
+
+	while (1) {

```

Comment:
I think it would fit better with the rest of the style if we did TRUE here.

Reply:
That's not very common. Most people either use while(1) or while(true). We shouldn't use the latter because most of the code is technically C (not C++) which doesn't know what "true" is.

Reply:
`while(true)` might be common these days, since `true` is part of the C++ standard. Perhaps back in the days `while(TRUE)` was common.

Reply:
It wasn't because TRUE was non-standard. `for (;;)` was actually more common because it was the definitive infinite loop quoted in K&R, which everyone considered to be the C bible.